packages/block-editor/src/components/inspector-controls-tabs/styles-tab.js

```diff
@@ -3,19 +3,140 @@
  */
 import { PanelBody } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import BlockStyles from '../block-styles';
 import InspectorControls from '../inspector-controls';
 import { useBorderPanelLabel } from '../../hooks/border';
+import { useBlockSettings } from '../../hooks/utils';
+import { store as blockEditorStore } from '../../store';
+import { ColorEdit } from '../../hooks/color';
+import { ColorToolsPanel } from '../global-styles/color-panel';
+
+function SectionBlockControls( { blockName, clientId, contentClientIds } ) {
+	const settings = useBlockSettings( blockName );
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+	const [ defaultControls, setDefaultControls ] = useState( {
+		text: true,
+		background: true,
+		button: true,
+		heading: true,
+		caption: true,
```

Comment:
@scruffian I have this set to true because I can't get the caption to show. Before this dynamic commit, the caption color would show. It works well to dynamically show only the relevant colors, ie. it doesn't show the heading block color unless there's a heading, etc.

The state addition was also just to see if I could fix the caption part. We don't necessarily need state for this.